src/lib/pages/account-details/components/tables/transactions/index.tsx

```diff
@@ -0,0 +1,187 @@
+    walletAddress,
+    "",
+    filters,
+    onViewMore ? 5 : pageSize,
+    offset
+  );
+
+  const onPageChange = (nextPage: number) => {
+    refetchCount();
+    setCurrentPage(nextPage);
+  };
+
+  const onPageSizeChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    const size = Number(e.target.value);
+    refetchCount();
+    setPageSize(size);
+    setCurrentPage(1);
+  };
+
+  const templateColumnsStyle =
+    "180px 70px minmax(360px, 1fr) max(250px) max(70px)";
+
+  if (isLoading) return <Loading />;
+
+  if (!transactions?.length && filterSelected.length > 0)
+    return (
+      <Flex
+        mt="20px"
+        py="64px"
+        direction="column"
+        borderY="1px solid"
+        borderColor="pebble.700"
+      >
+        <EmptyState
+          image="https://assets.alleslabs.dev/illustration/search-empty.svg"
+          message="No past transaction matches found with your input."
+        />
+      </Flex>
+    );
+
+  if (!transactions?.length)
+    return (
+      <Flex
+        mt="20px"
+        py="64px"
+        direction="column"
+        borderY="1px solid"
+        borderColor="pebble.700"
+      >
+        <EmptyState message="This account did not submit any transactions before." />
+      </Flex>
+    );
+
+  return (
+    <>
+      <TableContainer>
+        <Grid templateColumns={templateColumnsStyle}>
+          <TableHeader>Transaction Hash</TableHeader>
+          <TableHeader />
+          <TableHeader>Messages</TableHeader>
+          <TableHeader>Timestamp</TableHeader>
+          <TableHeader />
+        </Grid>
+        {transactions.map((transaction) => (
+          <TxsTableRow
+            key={transaction.hash}
+            transaction={transaction}
+            templateColumnsStyle={templateColumnsStyle}
+          />
+        ))}
+      </TableContainer>
+      {totalData &&
+        (onViewMore
+          ? totalData > 5 && <ViewMore onClick={onViewMore} />
+          : totalData > 10 && (
+              <Pagination
+                currentPage={currentPage}
+                pagesQuantity={pagesQuantity}
+                offset={offset}
+                totalData={totalData}
+                scrollComponentId={scrollComponentId}
+                pageSize={pageSize}
+                onPageChange={onPageChange}
+                onPageSizeChange={onPageSizeChange}
+              />
+            ))}
+    </>
+  );
+};
+
+export const TransactionTable = (
+  transactionTableProps: TransactionTableProps
+) => {
+  const [filters, setFilters] = useState<TxFilters>(DEFAULT_FILTERS);
+
+  const handleSetFilters = (filter: string, bool: boolean) => {
+    setFilters((prevFilters) => ({ ...prevFilters, [filter]: bool }));
+  };
+
+  const filterSelected = useMemo(() => {
+    return Object.keys(filters).reduce((acc: string[], key: string) => {
+      if (filters[key as keyof typeof filters]) {
+        acc.push(key);
+      }
+      return acc;
+    }, []);
+  }, [filters]);
```

Comment:
- remove useMemo as the filters dep will always update anyway
- can do something like 
```
Object.keys(filters).filter(key => filters[key])
```